zio-json/shared/src/main/scala/zio/json/ast/ast.scala

```diff
@@ -43,14 +43,12 @@ sealed abstract class Json { self =>
 
   override final def equals(that: Any): Boolean = {
     def objEqual(left: Map[String, Json], right: Chunk[(String, Json)]): Boolean =
-      if (right.isEmpty) true
-      else
-        right.find { case (key, r) =>
-          left.get(key) match {
-            case Some(l) if l != r => true
-            case _                 => false
-          }
-        }.isEmpty
+      right.forall { case (key, r) =>
+        left.get(key) match {
+          case Some(l) if l == r => true
+          case _                 => false
+        }
+      }
```

Comment:
If I understand this correctly, this ensures that all values of `right` are present in `left`, but does would still return true if `left` has additional fields not represented in `right`. Is that the expected behavior? In the attached tests, the one way check exists: 
```scala
          val obj1 = Json.Obj(
            "quux" -> Json.Str("1"),
            "bar"  -> Json.Str("2")
          )

          val obj2 = Json.Obj(
            "quux" -> Json.Str("1"),
            "bar"  -> Json.Str("2"),
            "maux" -> Json.Str("3")
          )

          assertTrue(obj1 != obj2)
```
but not the reflective
```scala
          assertTrue(obj2 != obj1)
```